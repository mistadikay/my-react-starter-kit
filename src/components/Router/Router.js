import React from 'react';
import { browserHistory, Router, Route, IndexRoute } from 'react-router';

import App from '~/components/App';
import Login from '~/components/Login';
import Dashboard from '~/components/Dashboard';
import Editor from '~/components/Editor';
import EmptyPage from '~/components/EmptyPage';

type Props = {
  onUpdate: Function
};

export default class extends React.Component {
  props: Props
  static defaultProps = {
    onUpdate() { /* do nothing */ }
  }

  render() {
    return (
      <Router history={ browserHistory } onUpdate={ this.props.onUpdate }>
        <Route name="ADMIN" path="/" component={ App }>
          <IndexRoute component={ Login } />
          <Route component={ Dashboard }>
            <Route path="editor" component={ Editor } />
            <Route path="*" component={ EmptyPage } />
          </Route>
        </Route>
      </Router>
    );
  }
}
