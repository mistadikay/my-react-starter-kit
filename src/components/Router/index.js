import { connect } from 'react-redux';
import Component from './Router';

const mapDispatchToProps = () => ({
  onUpdate() {
    // we can dispatch route change to use it in redux
    // `this.state` is coming from `Router` component
    // see https://github.com/ReactTraining/react-router/issues/1539#issuecomment-121499082
    // dispatch(changeRoute(this.state));
  }
});

export default connect(
  null,
  mapDispatchToProps
)(Component);
