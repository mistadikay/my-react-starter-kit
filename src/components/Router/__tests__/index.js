/* eslint-env es6, jest */
import React from 'react';
import { shallow } from 'enzyme';

import Router from '../Router';

test('Router', () => {
  const component = shallow(
    <Router />
  );

  expect(component).toMatchSnapshot();
});
