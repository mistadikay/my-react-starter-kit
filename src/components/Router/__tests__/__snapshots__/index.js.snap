// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Router
    history={
        Object {
            "createHref": [Function],
            "createKey": [Function],
            "createLocation": [Function],
            "createPath": [Function],
            "getCurrentLocation": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "listen": [Function],
            "listenBefore": [Function],
            "push": [Function],
            "replace": [Function],
            "transitionTo": [Function],
          }
    }
    onUpdate={[Function]}
    render={[Function]}
>
    <Route
        component={[Function]}
        name="ADMIN"
        path="/"
    >
        <IndexRoute
            component={[Function]}
        />
        <Route
            component={[Function]}
        >
            <Route
                component={[Function]}
                path="editor"
            />
            <Route
                component={[Function]}
                path="*"
            />
        </Route>
    </Route>
</Router>,
  "nodes": Array [
    <Router
      history={
            Object {
                  "createHref": [Function],
                  "createKey": [Function],
                  "createLocation": [Function],
                  "createPath": [Function],
                  "getCurrentLocation": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "listen": [Function],
                  "listenBefore": [Function],
                  "push": [Function],
                  "replace": [Function],
                  "transitionTo": [Function],
                }
      }
      onUpdate={[Function]}
      render={[Function]}
>
      <Route
            component={[Function]}
            name="ADMIN"
            path="/"
      >
            <IndexRoute
                  component={[Function]}
            />
            <Route
                  component={[Function]}
            >
                  <Route
                        component={[Function]}
                        path="editor"
                  />
                  <Route
                        component={[Function]}
                        path="*"
                  />
            </Route>
      </Route>
</Router>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <_class
        onUpdate={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": _class {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onUpdate": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Router
          history={
                    Object {
                              "createHref": [Function],
                              "createKey": [Function],
                              "createLocation": [Function],
                              "createPath": [Function],
                              "getCurrentLocation": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "listen": [Function],
                              "listenBefore": [Function],
                              "push": [Function],
                              "replace": [Function],
                              "transitionTo": [Function],
                            }
          }
          onUpdate={[Function]}
          render={[Function]}
>
          <Route
                    component={[Function]}
                    name="ADMIN"
                    path="/"
          >
                    <IndexRoute
                              component={[Function]}
                    />
                    <Route
                              component={[Function]}
                    >
                              <Route
                                        component={[Function]}
                                        path="editor"
                              />
                              <Route
                                        component={[Function]}
                                        path="*"
                              />
                    </Route>
          </Route>
</Router>,
        "_debugID": 2,
        "_renderedOutput": <Router
          history={
                    Object {
                              "createHref": [Function],
                              "createKey": [Function],
                              "createLocation": [Function],
                              "createPath": [Function],
                              "getCurrentLocation": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "listen": [Function],
                              "listenBefore": [Function],
                              "push": [Function],
                              "replace": [Function],
                              "transitionTo": [Function],
                            }
          }
          onUpdate={[Function]}
          render={[Function]}
>
          <Route
                    component={[Function]}
                    name="ADMIN"
                    path="/"
          >
                    <IndexRoute
                              component={[Function]}
                    />
                    <Route
                              component={[Function]}
                    >
                              <Route
                                        component={[Function]}
                                        path="editor"
                              />
                              <Route
                                        component={[Function]}
                                        path="*"
                              />
                    </Route>
          </Route>
</Router>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <_class
    onUpdate={[Function]}
/>,
}
`;
